app/classes/query/scope_modules/sequence_operators.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# methods for moving forward/back up/down in Query results
+module Query::ScopeModules::SequenceOperators
+  # Current place in results, as an id.  (Returns nil if not set yet.)
+  attr_reader :current_id
+
+  # Set current place in results; takes id (String or Integer).
+  def current_id=(id)
+    @save_current_id = @current_id = id.to_s.to_i
+  end
+
+  # Reset current place in results to place last given in a "current=" call.
+  def reset
+    @current_id = @save_current_id
+  end
+
+  # Current place in results, instantiated.  (Returns nil if not set yet.)
+  def current(*)
+    @current_id ? instantiate_results([@current_id], *).first : nil
+  end
+
+  # Set current place in results; takes instance or id (String or Integer).
+  def current=(arg)
+    if arg.is_a?(model)
+      @results ||= {}
+      @results[arg.id] = arg
+      self.current_id = arg.id
+    else
+      self.current_id = arg
+    end
+  end
+
+  # Move to first place.
+  def first
+    new_self = self
+    id = new_self.result_ids.first
+    if id.positive?
+      @current_id = id
+    else
+      new_self = nil
+    end
+    new_self
```

Comment:
Will this work?  Regardless, do you need safe navigation in `.result_ids.first`
```suggestion
    if result_ids.first.positive?
      @current_id = id
    else
      nil
    end
```

Reply:
Good question. This is very old code and I didn't want to touch it. 

With almost all of Jason's logic, when i found apparent quirks like this i tend to discover after messing with it for 3 hours that i cannot improve on it. That happened so many times i got scared to waste time trying.

Here I'd bet that the object needs to be cloned for something else to work, but idk.